### DISCUSSION TITLE
Ask questions and get answers here - June 2021

### DISCUSSION THREAD

**Discussion by ghobona** [opengeospatial/developer-track#18] [Q&A]
Ask questions on this channel.

**Comment by bretmulvey** [comment]
For GeoPose JSON, in some examples it looks like some key elements require a second, non-JSON, parser, e.g. [here](https://github.com/opengeospatial/GeoPose/blob/main/Data/Examples/SolarPose/GeoPose.Composite.Chain.Instance.json) to get at the outer position, relative position, and orientation data using a querystring-style parser individual JSON strings. Has there been discussion to require these elements to be exposed directly at the JSON level?

**Reply by 3DXScape** [reply]
For external authorities defining a reference frame, we chose a three-element model: authority, id, parameters. The parameter encoding is defined by the external authority. It might be a chunk of URL encoded JSON or a query string key=value collection as in this example. This has the advantage of flexibility but requires some extra information in order to create/interpret. The problem addressed is that different authorities have wildly different ways of specifying reference frames. Does this make sense?

**Reply by bretmulvey** [reply]
It makes total sense. Is this is considered a potential issue, or not because this is part of "advanced"?

**Reply by 3DXScape** [reply]
It is a practical issue even if not a GitHub issue :-) We probably need to provide some informative guidance and examples for common ways of specifying reference frames - OGC/ISO WKT, NASA SPICE, EPSG, etc and how this can/could be/should be be encoded in the parameters element.
